src/main/java/org/opentripplanner/middleware/controllers/api/MonitoredTripController.java

```diff
@@ -110,11 +138,64 @@ private void processTripQueryParams(MonitoredTrip monitoredTrip, Request req) {
 
     @Override
     MonitoredTrip preUpdateHook(MonitoredTrip monitoredTrip, MonitoredTrip preExisting, Request req) {
-        preCreateOrUpdateChecks(monitoredTrip, req);
+        // Wait for any existing CheckMonitoredTrip jobs to complete before proceeding
+        if (monitoredTripLocks.containsKey(monitoredTrip)) {
+            int maxWaitTimeMillis = 4000;
+            int timeWaitedMillis = 0;
+            do {
+                try {
+                    Thread.sleep(500);
+                } catch (InterruptedException e) {
+                    logMessageAndHalt(
+                        req,
+                        HttpStatus.INTERNAL_SERVER_ERROR_500,
+                        "The trip analyzer prevented the update of this trip. Please try again."
```

Comment:
I'm worried this message won't make much sense to users. How about: "You attempted to save this trip while a trip monitor check was in progress. Please try again in a moment."

Reply:
Changed wording in 6948fab3bf5ead8b87a4b47676a38fe9a6e1691f